src/config.ts

```diff
@@ -248,7 +250,29 @@ export class KubeConfig implements SecurityAuthentication {
         agentOptions.passphrase = httpsOptions.passphrase;
         agentOptions.rejectUnauthorized = httpsOptions.rejectUnauthorized;
 
-        context.setAgent(new https.Agent(agentOptions));
+        let agent: https.Agent | SocksProxyAgent | HttpProxyAgent | HttpsProxyAgent;
+
+        if (cluster && cluster.proxyUrl) {
+            if (cluster.proxyUrl.startsWith('socks')) {
+                agent = new SocksProxyAgent(cluster.proxyUrl, agentOptions);
+            } else if (cluster.proxyUrl.startsWith('http')) {
+                agent = new HttpsProxyAgent({
+                    proxy: cluster.proxyUrl,
+                    ...agentOptions,
```

Comment:
Is the spread operator being used here and a few lines down only to appease the TypeScript compiler? If so, could it be rewritten to avoid the spread operator?

Reply:
You could set the `proxy` property on the `agentOptions` and so avoid using the spread operator.

i.e.

```
agentOptions.proxy = cluster.proxyUrl
...

agent = new HttpProxyAgent(agentOptions);
```

Reply:
I rewrote it like this:
```typescript
  const httpsProxyAgentOptions: HttpsProxyAgentOptions = agentOptions as HttpsProxyAgentOptions;
  httpsProxyAgentOptions.proxy = cluster.proxyUrl;
  agent = new HttpsProxyAgent(httpsProxyAgentOptions);
``` 
so the compiler is happy and there is no spread operator.